tests/test_crawlers.py

```diff
@@ -1,17 +1,41 @@
 """Test suite for crawlers"""
-#pylint: disable=protected-access
 
 import logging
 import os
 import unittest
 import unittest.mock as mock
 from datetime import datetime, timezone
+#import geospaas_harvesting.harvest as harvest
 
 import requests
 
 import geospaas_harvesting.crawlers as crawlers
 
 
+class WebDirectoryCrawlerExceptionTestCase(unittest.TestCase):
+    """Tests for the webDirectory crawler Exceptions"""
+
+    def test_lack_of_definition_of_get_download_url(self):
+        """shall return "NotImplementedError" exception in the case of lack of definition of get_download_url method """
+        my_harvester = crawlers.WebDirectoryCrawler
+        with self.assertRaises(NotImplementedError):
+            my_harvester.get_download_url("test_text", "test_text2")
+
+    def test_lack_of_definition_of_set_initial_state(self):
+        """shall return "NotImplementedError" exception in the case of lack of definition of set_initial_state method """
+        with self.assertRaises(NotImplementedError):
+            crawlers.Crawler.set_initial_state(self)
```

Comment:
Same here, don't invoke an instance method directly from the class.
And why would you pass it `self`? In this context `self` refers to the current instance of `WebDirectoryCrawlerExceptionTestCase`

Reply:
This is not my test. It was written by you. COuld you please explain?

Reply:
I did not write that.
Same thing as before, you should instantiate a crawler and then use it.

Reply:
Done in the next commit.